examples/test_spectrum_model.py

```diff
@@ -1,47 +1,74 @@
 """Spectrum model API for Gammapy.
 """
+import numpy as np
 from astropy.units import Quantity as Q
 from gammapy.spectrum.models import (
     SpectralModel, PowerLaw, ExponentialCutoffPowerLaw
 )
+from gammapy.irf import (
+    EffectiveAreaTable,
+    EnergyDispersion
+)
+from gammapy.utils.energy import EnergyBounds
+
+
+# TODO clean up and add to gammapy.irf
```

Comment:
Did you see this?
https://gammapy.readthedocs.io/en/latest/api/gammapy.irf.EnergyDispersion.html#gammapy.irf.EnergyDispersion.from_gauss

Maybe you could check / fix that Gaussian energy resolution and add this "perfect" resolution as a separate similar method?
I'm never quite sure where to put such "make example something" functions ... but maybe classmethods are best?